ipykernel/kernelbase.py

```diff
@@ -1274,3 +1413,6 @@ async def _at_shutdown(self):
                     ident=self._topic("shutdown"),
                 )
                 self.log.debug("%s", self._shutdown_message)
+
+    def _supports_kernel_subshells(self):
```

Comment:
```suggestion
    def _supports_kernel_subshells(self) -> bool:
```

Reply:
And is it worth making it a property ? (I don't really care as it's private) I guess the questions is wether shell_channel_thread may change ?

Reply:
I think making it a property is sensible. I cannot imagine any circumstances in which `shell_channel_thread` may change, but you never know...

Reply:
Made it a property in a1fb1de.